src/MiniProfiler.AspNetCore/MiniProfilerMiddleware.cs

```diff
@@ -36,11 +36,11 @@ public class MiniProfilerMiddleware
         public MiniProfilerMiddleware(
             RequestDelegate next,
             IHostingEnvironment hostingEnvironment,
-            MiniProfilerOptions options)
+            IOptions<MiniProfilerOptions> options)
         {
             _next = next ?? throw new ArgumentNullException(nameof(next));
             _env = hostingEnvironment ?? throw new ArgumentNullException(nameof(hostingEnvironment));
-            Options = options ?? throw new ArgumentNullException(nameof(options));
+            Options = options.Value ?? throw new ArgumentNullException(nameof(options));
```

Comment:
I commented on this elsewhere about potentially changing the registration of the options to not use `Configure()` and `IOptions<>` if none of the options actually depend on pulling information from a configuration source. But it's worth pointing out that should you not decide to do this and instead keep `Configure`/`IOptions`, that you should not save the result of the `Value` property beyond it's use. This is what allows your options to be refreshed should the source configuration be notified of a change. By saving the value, you can run into a situation where your internal Options property in this class will be what it was when the app first started, but requesting `IOptions<MiniProfilerOptions>` again from the ServiceProvider yields an options instance with Value being a different instance entirely.

Reply:
Thanks for the tips here, I think I addressed both in latest.

We *may* have something loading from JSON earlier (e.g. a storage configed location), so that's the reason for `.Configure()` now (in relation to your other comment).